soc/arm/nuvoton_npcx/common/CMakeLists.txt

```diff
@@ -1,9 +1,10 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_include_directories(.)
-
+zephyr_library_sources_ifdef(CONFIG_PM power.c)
```

Comment:
Why is power.c added to ZEPHYR_CURRENT_LIBRARY, while scfg.c and registers.c are added to the main "zephyr" library?

Reply:
power.c is considered only if SOC_POWER_MANAGEMENT is enabled. Both scfg.c and register.c are basic utilities used by all npcx series. Do I miss something?

Reply:
I don't understand why you are using zephyr_library_sources_ifdef() instead of zephyr_sources_ifdef() for power.c.

zephyr_sources_ifdef() adds the object file into libzephyr.a, while zephyr_library_sources_ifdef() adds the object file into library specified by the ZEPHYR_CURRENT_LIBRARY variable.

Reply:
Actually, both 'zephyr_sources_ifdef(CONFIG_PM ...)' and 'zephyr_library_sources_ifdef(CONFIG_PM ...)' are found in CMakeLists.txt to add power management implementation. But the latter is chosen by more chip vendors and subsys/pm. Will modify it later.

Reply:
Done.